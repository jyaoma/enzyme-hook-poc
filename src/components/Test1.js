import React, { useState } from 'react';

const Test1 = ({ testProp }) => {
  // console.error(testProp);
  const [ toggleChange, setToggleChange ] = useState(false);

  return <div testProp={testProp}>
    <input className="my-checkbox" type="checkbox" onChange={() => setToggleChange(!toggleChange)} checked={toggleChange} value="oof" />
    <div>{toggleChange ? 'True' : 'False' }</div>
  </div>
};

export default Test1;
