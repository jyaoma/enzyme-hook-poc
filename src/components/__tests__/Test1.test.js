import React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';

import Test1 from '../Test1';

describe('Test1', () => {
  it('should render correctly', () => {
    const rendered = renderer.create(<Test1 />);
    expect(rendered.toJSON()).toMatchSnapshot();
  });

  it('can read props', () => {
    const tree = shallow(<Test1 testProp={'oof'} />);
    expect(tree.props().testProp).toEqual('oof');
  });

  // xit('can read state', () => {
  //   const tree = mount(<Test1 testProp={'oof'} />);
  //   // console.warn(tree.state());
  //   console.warn(tree.debug());
  // });

  it('can find stuff inside', () => {
    const tree = mount(<Test1 testProp={'oof'}/>);
    const checkbox = tree.find('.my-checkbox');
    expect(checkbox.props().checked).toBe(false);
  });

  it('can make changes', () => {
    const tree = mount(<Test1 testProp={'oof'}/>);
    const checkbox = tree.find('.my-checkbox');
    checkbox.simulate('change');

    tree.update();
    expect(tree.find('.my-checkbox').props().checked).toBe(true);
  });
});