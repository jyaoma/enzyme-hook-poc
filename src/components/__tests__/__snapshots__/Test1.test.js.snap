// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test1 should render correctly 1`] = `
<div>
  <input
    checked={false}
    className="my-checkbox"
    onChange={[Function]}
    type="checkbox"
    value="oof"
  />
  <div>
    False
  </div>
</div>
`;
